elevation_mapping/src/sensor_processors/SensorProcessorBase.cpp

```diff
@@ -158,5 +163,37 @@ void SensorProcessorBase::removePointsOutsideLimits(
   ROS_DEBUG("removePointsOutsideLimits() reduced point cloud to %i points.", (int) pointClouds[0]->size());
 }
 
+bool SensorProcessorBase::cleanPointCloud(const pcl::PointCloud<pcl::PointXYZRGB>::Ptr pointCloud)
+{
+	pcl::PassThrough<pcl::PointXYZRGB> passThroughFilter;
+	pcl::PointCloud<pcl::PointXYZRGB> tempPointCloud;
+
+  // remove nan points
+  std::vector<int> indices;
+  pcl::removeNaNFromPointCloud(*pointCloud, tempPointCloud, indices);
+  tempPointCloud.is_dense = true;
+  pointCloud->swap(tempPointCloud);
+
+  // cutoff points with z values
+  passThroughFilter.setInputCloud(pointCloud);
+  passThroughFilter.setFilterFieldName("z");
+  passThroughFilter.setFilterLimits(sensorParameters_.at("cutoff_min_depth"),
+                                    sensorParameters_.at("cutoff_max_depth"));
+  passThroughFilter.filter(tempPointCloud);
+  pointCloud->swap(tempPointCloud);
```

Comment:
Moving the pass through filter to the sensor processor base class implies that it should be used for all sensor data (not only RealSense). Is that the case? If yes, do cutoff_min/max_depth need to be added to the other sensor processor configuration files?

Reply:
Yes, I think this filter would be the same for all sensors.
Because I didn't want to change the behavior of the other sensor now, I didn't change the other sensor's code. but, if you agree, I can also change the other sensors.

Reply:
This is nasty, why did you move this into the base class and changed from pure virtual to virtual? If you want to add a set of generic filters, this should be done as a collection of filter that can be applied and reordered for each sensor.

Reply:
I think the filtering functions are the same for all sensor types. This is why I moved to the base class.
Do you think that the sensor base class should only hold each filtering functions (NaN, cutoff, voxelgrid) and each sensor should implement `cleanPointCloud` function to call these functions?